torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -642,6 +642,208 @@ class IrParser {
+            std::vector<int> output_mask;
+            for (const auto value : out_mask_list->vec()) {
+              output_mask.emplace_back(static_cast<int>(value));
+            }
+
+            const size_t kOuterNumDims = input->nDims() - kNormShapeNumDims;
+
+            std::vector<int> outer_reduction_axes(kOuterNumDims);
+            std::vector<bool> outer_broadcast_mask(input->nDims(), false);
+            for (size_t idx = 0; idx < kOuterNumDims; ++idx) {
+              outer_reduction_axes[idx] = idx;
+              outer_broadcast_mask[idx] = true;
+            }
+
+            std::vector<int> inner_reduction_axes(kNormShapeNumDims);
+            std::vector<bool> inner_broadcast_mask(input->nDims(), false);
+            Val* num_features = nullptr;
+            for (int idx = 0; idx < kNormShapeNumDims; ++idx) {
+              const size_t axis = input->nDims() - 1 - idx;
+              inner_reduction_axes[idx] = axis;
+              inner_broadcast_mask[axis] = true;
+              num_features = (num_features == nullptr)
+                  ? input->domain()->domain()[axis]->extent()
+                  : mul(num_features,
+                        input->domain()->domain()[axis]->extent());
+            }
+
+            // TODO: grad_bias and grad_weight are disabled because
+            // they are incompabilble with grad_in fusion
+            // Requires seperate kernels
+
+            /*
+            auto grad_bias = sum(grad_out, outer_reduction_axes);
+            fusion.addOutput(grad_bias);
+
+            auto bcast_mean = broadcast(mean, inner_broadcast_mask);
+            auto bcast_rstd = broadcast(rstd, inner_broadcast_mask);
+            auto x_hat = mul(sub(input, bcast_mean), bcast_rstd);
+            auto grad_weight = sum(mul(grad_out, x_hat), outer_reduction_axes);
+            fusion.addOutput(grad_weight);
+            */
+
+            // auto bcast_mean = broadcast(mean, inner_broadcast_mask);
+            // auto bcast_rstd = broadcast(rstd, inner_broadcast_mask);
+            // auto x_hat = mul(sub(input, bcast_mean), bcast_rstd);
+            auto x_hat = mul(sub(input, mean), rstd);
+
+            TensorView* grad_x_hat = nullptr;
+            if (weight == nullptr) {
+              grad_x_hat = grad_out;
+            } else {
+              auto* bcast_weight = broadcast(weight, outer_broadcast_mask);
+              grad_x_hat = mul(grad_out, bcast_weight);
+            }
+
+            auto* a = mul(num_features, grad_x_hat);
+
+            auto* b = sum(grad_x_hat, inner_reduction_axes);
+            auto* bcast_b = broadcast(b, inner_broadcast_mask);
+
+            auto* c1 = mul(grad_x_hat, x_hat);
+            auto* c2 = sum(c1, inner_reduction_axes);
+            auto* bcast_c2 = broadcast(c2, inner_broadcast_mask);
+            auto* c3 = mul(x_hat, bcast_c2);
+
+            auto* inner = sub(sub(a, bcast_b), c3);
+
+            auto reciprocal_size = div(new Float(1), num_features);
```

Comment:
nitpick, we do have a unary op with `UnaryOpType::Reciprocal`

Reply:
It tried it and it didn't work. When using `reciprocal`, the gradient becomes zero.